### ISSUE TITLE
Diagnostics error in Pyright LSP for `list.append` method and `dict.update` when using Neovim or Zed

### ISSUE THREAD

**Issue by codestantine** [microsoft/pyright#9480]
### Diagnostics error in Pyright LSP for `list.append` and `dict.update` methods when using Neovim with Kickstart setup and also with Zed editor

---

### Environment
- **Operating System**: Ubuntu 24
- **Neovim Version**: 0.10.2
- **Neovim Setup**: Kickstart configuration
- **LSP Server**: Pyright
- **Language client log**: `/home/amir/.local/state/nvim/lsp.log`
- **LSP Client**: Pyright (configured via Mason)
- **Filetype**: Python

---

### Steps to Reproduce (list.append)
1. Open a new Python file in Neovim or Zed.
2. Write the following code:
   ```python
   a = list("hello")
   a.append(100)
   print(a)
   ```

### Expected Behavior
The code should not trigger a diagnostics error because list.append is expected to accept any object type, including integers, as a valid element for a Python list.

### Actual Behavior
A diagnostics error is triggered, suggesting that an integer (100) cannot be assigned to a list where elements are expected to be of type str.

### Diagnostics Error
Argument of type "Literal[100]" cannot be assigned to parameter "object" of type "str" in function "append" "Literal[100]" is not assignable to "str [reportArgumentType]

### Steps to Reproduce (dict.update)
1. Open a new Python file in Neovim/Zed.
2. Write the following code:
   ```python
   d = dict(zip("hello", range(5)))
   d.update({"another": "value"})
   print(f"d: {d}")
   ```
### Expected Behavior
The code should not trigger a diagnostics error because dict.update is expected to accept a dictionary with key-value pairs of any type.

### Actual Behavior
A diagnostics error is triggered, suggesting that the provided dictionary {"another": "value"} cannot be assigned to the expected parameter type.

### Diagnostics Error
1. No overloads for "update" match the provided arguments [reportCallIssue]
2. Argument of type "dict[str, str]" cannot be assigned to parameter "m" of type "Iterable[tuple[str, int]]" in function "update"
   "Literal['another']" is not assignable to "tuple[str, int]" [reportArgumentType]

---

### Steps to Reproduce (list.append)
1. Open a new Python file in Neovim or Zed.
2. Write the following code:
   ```python
   a = list("hello")
   a.append(100)
   print(a)

### Expected Behavior
The code should not trigger a diagnostics error because list.append is expected to accept any object type, including integers, as a valid element for a Python list.

### Actual Behavior
A diagnostics error is triggered, suggesting that an integer (100) cannot be assigned to a list where elements are expected to be of type str.

### Diagnostics Error
Argument of type "Literal[100]" cannot be assigned to parameter "object" of type "str" in function "append"
  "Literal[100]" is not assignable to "str" [reportArgumentType]

### Steps to Reproduce (dict.update)
Open a new Python file in Neovim or Zed.
Write the following code:
```
d = dict(zip("hello", range(5)))
d.update({"another": "value"})
print(f"d: {d}")
```

### Expected Behavior
The code should not trigger a diagnostics error because dict.update is expected to accept a dictionary with key-value pairs of any type.

### Actual Behavior
A diagnostics error is triggered, suggesting that the provided dictionary {"another": "value"} cannot be assigned to the expected parameter type.

### Diagnostics Error
No overloads for "update" match the provided arguments [reportCallIssue]
Argument of type "dict[str, str]" cannot be assigned to parameter "m" of type "Iterable[tuple[str, int]]" in function "update"
  "Literal['another']" is not assignable to "tuple[str, int]" [reportArgumentType]


**VS Code extension or command-line**
The code works as expected in VSCode but triggers the mentioned problems in Neovim and Zed.

### Additional Comments
If any additional information is needed, I’m happy to provide it. I’m not a professional developer, so if there's something I missed, I appreciate your guidance.
